packages/singleton-manager/README.md

```diff
@@ -0,0 +1,297 @@
+
+```js
+// my-singleton.js
+import { singletonManager } from 'singleton-manager';
+import { MySingleton } from './src/MySingleton.js';
+
+export const overlays =
+  singletonManager.get('my-singleton/my-singleton.js::1.x') || new MySingleton();
+```
+
+## Convention Singleton Key
+
+The key for a singleton needs to be "unique" for the package.
+Hence the following convention helps maintaining this.
+
+As a key use the `<package>::<unique-variable>::<semver-range>`.
+
+Examples Do:
+
+- `overlays::overlays::1.x` - instance created in index.js
+- `@scope/overlays::overlays::1.x` - with scope
+- `overlays::overlays::1.x` - version 1.x.x (> 1.0.0 you do 1.x, 2.x)
+- `overlays::overlays::2.x` - version 2.x.x (> 1.0.0 you do 1.x, 2.x)
+- `overlays::overlays::0.10.x` - version 0.10.x (< 1.0.0 you do 0.1.x, 0.2.x)
+
+Examples Don't:
+
+- `overlays` - too generic
+- `overlays::overlays` - you should include a version
+- `overlays::1.x` - you should include a package name & unique var
+- `./index.js::1.x` - it should start with a package name
+
+---
+
+## Singleton Manager Rational
+
+We have an app with 2 pages.
+
+- page-a uses overlays 1.x
+- page-b uses overlays 2.x (gets installed nested)
+
+```txt
+my-app (node_modules)
+├── overlays (1.x)
+├── page-a
+│   └── page-a.js
+└── page-b
+    ├── node_modules
+    │   └── overlays (2.x)
+    └── page-b.js
+```
+
+The tough part in this case is the OverlaysManager within the overlays package as it needs to be a singleton.
+
+It starts of simplified like this
+
+```js
+export class OverlaysManager {
+  name = 'OverlayManager 1.x';
+  blockBody = false;
+  constructor() {
+    this._setupBlocker();
+  }
+  _setupBlocker() {
+    /* ... */
+  }
+  block() {
+    this.blockBody = true; // ...
+  }
+
+  unBlock() {
+    this.blockBody = false; // ...
+  }
+}
+```
+
+## Example A (fail)
+
+See it "fail" e.g. 2 separate OverlaysManager are at work and are "fighting" over the way to block the body.
+
+```bash
+npm run start:fail
+```
+
+Steps to reproduce:
+
+1. Page A click on block
+2. Page B => "Blocked: false" (even when hitting the refresh button)
+
+See [the code](./demo/fail/demo-app.js).
+
+---
+
+## Example B (singleton manager)
+
+The breaking change in `OverlayManager` was renaming of 2 function (which has been deprecated before).
+
+- `block()` => `blockingBody()`
+- `unBlock()` => `unBlockingBody()`
+
+knowing that we can create a Manager that is compatible with both via
+
+```js
+import { OverlaysManager } from 'overlays';
+
+class CompatibleManager extends OverlaysManager {
+  blockingBody() {
+    this.block();
+  }
+  unBlockingBody() {
+    this.unBlock();
+  }
+}
+```
+
+all that is left is a to "override" the default instance of the "users"
+
+```js
+import { singletonManager } from 'singleton-manager';
+
+const manager = new CompatibleManager();
+singletonManager.set('overlays::overlays::1.x', compatibleManager);
+singletonManager.set('overlays::overlays::2.x', compatibleManager);
```

Comment:
```suggestion
const compatibleOverlaysManager = new CompatibleOverlaysManager();
singletonManager.set('overlays::overlays::1.x', compatibleOverlaysManager);
singletonManager.set('overlays::overlays::2.x', compatibleOverlaysManager);
```

Reply:
Makes it a bit easier to read the docs for me, because manager is a little too generic

Reply:
👍